src/styles/blocks/header.scss

```diff
@@ -0,0 +1,69 @@
+.header {
+  display: flex;
+  justify-content: space-between;
+  align-items: center;
+  box-shadow: 0 2px 4px rgba(0, 0, 0, 0.05);
+  padding: 0 50px;
+
+  &__link {
+    display: flex;
+    align-items: center;
+  }
+
+  &__nav {
+    display: flex;
+    align-items: center;
+    height: 60px;
+
+    &-list {
+      display: flex;
+      list-style: none;
+      padding: 0;
+      height: 100%;
+    }
+
+    &-item {
+      position: relative;
+      height: 100%;
+      margin: 0 10px;
+    }
+
+    &-link {
+      display: flex;
+      align-items: center;
+      text-decoration: none;
+      height: 100%;
+      font-size: $font-size-medium;
+      color: $secondary-color;
+      text-transform: uppercase;
+      transition-duration: 300ms;
+      font-weight: $font-weight-medium;
+
+      &:hover {
+        color: $primary-color;
+      }
+    }
+
+    &-item:first-child {
+      margin-left: 0;
+    }
+
+    &-item:last-child {
+      margin-right: 0;
+    }
```

Comment:
The `:first-child` and `:last-child` pseudo-classes are well used to remove the margin from the first and last items. However, ensure that this doesn't lead to any unwanted side effects when the header is viewed on different screen sizes or when items are added or removed. It's a good practice to check the design's responsiveness and adaptability.